status-im/status-react
# Add send logs command

[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #6710 

### Summary:

[comment]: # (Summarise the problem and how the pull request solves it)

This PR adds one-click log export for Android, Windows, Linux, and MacOS. iOS is not implemented as I don't have the hardware required for it, but it will be just a question of implementing the `sendLogs` React method.

Sample zip file: [Status-debug-logs.zip](https://github.com/status-im/status-react/files/2702608/Status-debug-logs.zip)

Android UI:
![screenshot_20181221-123931](https://user-images.githubusercontent.com/138074/50341140-00ace780-051e-11e9-8d8f-5d26dc3a81ea.jpg)

Exported `db.json`:

![image](https://user-images.githubusercontent.com/138074/50341191-35b93a00-051e-11e9-9369-9bd0245ae39d.png)

An error message is shown in case free space is below 20 MB:

![image](https://user-images.githubusercontent.com/138074/50340173-41a2fd00-051a-11e9-9afd-347bedaed0c7.png)

### Review notes (optional):


I've tested Android, Windows and Linux, so it's particularly important to test MacOS. Let me know if there's more info you'd like to be added to the zip file that would be useful for future debugging.

### Testing notes:

#### Platforms
- Android
- iOS (the new Send logs button should simply not show in iOS)
- macOS
- Linux
- Windows


### Steps to test:
- Open Status
- Sign in
- Go to Profile/Advanced
- Enable Development Mode
- Click Send Logs
- A Share operation should be started on mobile. A explorer window should be opened with the zip on Desktop.



### Next steps:

Remove https://status.im/build_status/status_go.html#Android-debugging section, as it won't be possible to access geth.log from `adb` anymore.

status: ready
